### ISSUE TITLE
We generate invalid code for varargs functions (test-case: rocksdb)

### ISSUE THREAD

**Issue by zhangjinpeng1987** [servo/rust-bindgen#361]
When I use `libbindgen` with rocksdb, code in bindings.rs doesn't work.

build.rs
```
extern crate libbindgen;

use std::env;
use std::path::PathBuf;

fn main() {
    // Tell Cargo to tell rustc to link the system rocksdb shared library.
    println!("cargo:rustc-link-lib=rocksdb");

    let bindings = libbindgen::Builder::default()
        .no_unstable_rust()
        .clang_arg("-std=c++11")
        .header("wrapper.hpp")
        .generate()
        .expect("Unable to generate bindings");
    let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
    bindings.write_to_file(out_path.join("bindings.rs")).expect("Couldn't write bindings!");
}
```
code snip in bindings.rs
```
impl std_namespace___any {
    #[inline]
    pub unsafe fn new(, ...) -> Self { // doesn't work
    ¦   let mut __bindgen_tmp = ::std::mem::uninitialized();
    ¦   std_namespace___any___any(&mut __bindgen_tmp);
    ¦   __bindgen_tmp
    }
}
```

**Comment by emilio** [comment]
Can you post your wrapper.hpp and your bindings.rs file?

We don't support varargs functions right now (and I don't think we can
support them properly), but we should skip generating code for them.

That's a known defect, but there are several things you can do to work
around this known defects when binding C++.

The unofficial guide is here:

 * https://github.com/servo/rust-bindgen/issues/342#issuecomment-267438457

But we should definitely clean that up and write up a user guide to how
to use bindgen effectively with C++, since it may not be as
straight-forward as binding to C libraries.

I'm leaving this open to track the varargs constructor (and functions in
general) issue.

Thanks for reporting this!

**Comment by zhangjinpeng1987** [comment]
@emilio Thanks for your response.  The following is my `wrapper.hpp` file
```
// bindgen-flags: -- -std=c++11

#ifndef WRAPPER_HPP
#define WRAPPER_HPP

#include <rocksdb/db.h>

#endif /* !WRAPPER_HPP */
```

**Comment by emilio** [comment]
This is now fixed, see #402

**Comment by fitzgen** [comment]
> This is now fixed, see #402
